gtk4/src/subclass/entry_buffer.rs

```diff
@@ -0,0 +1,233 @@
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Object as IsSubclassable<T>>::override_vfuncs(class);
+
+        let klass = class.as_mut();
+        klass.delete_text = Some(entry_buffer_delete_text::<T>);
+        klass.deleted_text = Some(entry_buffer_deleted_text::<T>);
+        klass.get_length = Some(entry_buffer_get_length::<T>);
+        klass.get_text = Some(entry_buffer_get_text::<T>);
+        klass.insert_text = Some(entry_buffer_insert_text::<T>);
+        klass.inserted_text = Some(entry_buffer_inserted_text::<T>);
+    }
+}
+
+unsafe extern "C" fn entry_buffer_delete_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.delete_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_deleted_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.deleted_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_get_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    n_bytes: *mut usize,
+) -> *const libc::c_char {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.get_text(wrap.unsafe_cast_ref(), &mut (n_bytes as usize))
+        .to_glib_full()
+}
+
+unsafe extern "C" fn entry_buffer_get_length<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.get_length(wrap.unsafe_cast_ref())
+}
+
+unsafe extern "C" fn entry_buffer_insert_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    charsptr: *const libc::c_char,
+    _n_chars: u32,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+    let chars: Borrowed<GString> = from_glib_borrow(charsptr);
+
+    imp.insert_text(wrap.unsafe_cast_ref(), position, &chars)
+}
+
+unsafe extern "C" fn entry_buffer_inserted_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    charsptr: *const libc::c_char,
+    _length: u32,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+    let chars: Borrowed<GString> = from_glib_borrow(charsptr);
+
+    imp.inserted_text(wrap.unsafe_cast_ref(), position, &chars)
```

Comment:
If `length == -1` then this is correct, otherwise you need to only pass the first `length` bytes (or characters? the docs are not exactly clear if they mean "character" or C `char` there...)

Reply:
Seems to be the number of characters per 
>  * Note that @n_chars is in characters, not in bytes.

Reply:
That doesn't make it easier to handle, but at least `std::str` has useful unicode API.

What does characters mean though? Same as a Rust `char`?

Reply:
I guess so, because it falls back to `g_utf8_strlen` in case the provided length is equal `< 0`